### PR TITLE
fix unittest for tx pool

### PR DESCRIPTION
**Pull request by shyun-comcom**
FIX unit test for TX pool (#696)
* remove `async` from `describe`
* `node` has only 10 txs, but test wants 11 txs
  * see https://github.com/ainblockchain/ain-blockchain/blob/master/unittest/tx-pool.test.js#L73
* `node1` -> `node`
* `Block.create()` arguments mismatched
* `initialNonce` should be started from `node.getNonce() + 1`

Related issues: 
- https://github.com/ainblockchain/ain-blockchain/issues/696

### CODE REVIEW

File: unittest/tx-pool.test.js [hunk 1]
```diff
@@ -621,15 +621,15 @@ describe('TransactionPool', async () => {
...
       const receipts = txsToDummyReceipts(transactions);
       const block = Block.create(
-          lastBlock.hash, [], transactions, receipts, number, lastBlock.epoch + 1, '',
-          node.account.address, []);
+          lastBlock.hash, [], {}, transactions, receipts, number, lastBlock.epoch + 1, '',
+          node.account.address, [], 0, 0);
       const newTransactions = {};
       newTransactions[node.account.address] = [];
-      let initialNonce = node.getNonce();
+      let initialNonce = node.getNonce() + 1;
```
Comment (platfowner) [inline unittest/tx-pool.test.js:632]: Why +1 needs to be added? Are you seeing what makes the difference between this line and line 38 above?

File: unittest/tx-pool.test.js [hunk 2]
```diff
@@ -621,15 +621,15 @@ describe('TransactionPool', async () => {
...
-      const lastBlock = node1.bc.genesisBlock;
-      const transactions = node1.tp.getValidTransactions();
+      const lastBlock = node.bc.genesisBlock;
+      const transactions = node.tp.getValidTransactions();
       const receipts = txsToDummyReceipts(transactions);
       const block = Block.create(
-          lastBlock.hash, [], transactions, receipts, number, lastBlock.epoch + 1, '',
-          node.account.address, []);
+          lastBlock.hash, [], {}, transactions, receipts, number, lastBlock.epoch + 1, '',
+          node.account.address, [], 0, 0);
```
Comment (liayoo) [inline unittest/tx-pool.test.js:629]: Thanks for updating this!!
Could you change the `[]` after `node.account.address` to `{}`, and add `Date.now()` at the very end?

File: unittest/tx-pool.test.js (hunk 1 above)
Comment (shyun-comcom) [reply unittest/tx-pool.test.js:632]: If not, tx pool has two txs with duplicated nonce.
Without `+1`, there are two transactions with nonce 0, and it makes test failed.
```
// node.tp.transactions[node.account.address]
[
  Transaction {
    tx_body: {
      operation: [Object],
      nonce: 0, <-----------------
      gas_price: 1,
      timestamp: 1638364264471
    },
    signature: '0xd904c9d7eaefefcc4d18703f9f5d7d0a638d304a090458bff9584bd7fff86a1a2c2e212ae5741b80f3040ca54b0f3010c960f0528b6dcc4bf948b678bb08bf98394d90a4e954e7aa4bd803cf61d50ab9f117f8688a1910d2e79ccd37dd5b94ce1c',
    hash: '0xd904c9d7eaefefcc4d18703f9f5d7d0a638d304a090458bff9584bd7fff86a1a',
    address: '0x00ADEc28B6a845a085e03591bE7550dd68673C1C',
    extra: { created_at: 1638364264474, executed_at: null }
  },
  Transaction {
    tx_body: {
      operation: [Object],
      nonce: 0, <-----------------
      gas_price: 1,
      timestamp: 1638364264477
    },
    signature: '0x0c42efb2c3b70e5b72d41b5daf4e38257e7b88133791fd50b601fbeefe083a591fef6301eb0511b905aef74b473524a57c1f8914754c2cd1698dd413fcfbb2cd1e9744fbe6f033f51c568d1adacbf4c5763af820694936e1f168b14a5e8da6851c',
    hash: '0x0c42efb2c3b70e5b72d41b5daf4e38257e7b88133791fd50b601fbeefe083a59',
    address: '0x00ADEc28B6a845a085e03591bE7550dd68673C1C',
    extra: { created_at: 1638364264480, executed_at: null }
  },
  ...
  Transaction {
    tx_body: {
      operation: [Object],
      nonce: 9,
      gas_price: 1,
      timestamp: 1638364264508
    },
    signature: '0x3afbae22ed3816ba75377ca37c5ad26ad9cf66e0d9fa10d8eeda7b66687e07bc9bd1667b207b0704c15d980330c7a7e22740f526136e04dee1716398b17bc5d40d9748575dfc3f03c96a4e9026799c8c4e29d8a957ba4fba89bd79d88e7cff521b',
    hash: '0x3afbae22ed3816ba75377ca37c5ad26ad9cf66e0d9fa10d8eeda7b66687e07bc',
    address: '0x00ADEc28B6a845a085e03591bE7550dd68673C1C',
    extra: { created_at: 1638364264511, executed_at: null }
  }
]
```

With `+1`, it works properly.
```
// node.tp.transactions[node.account.address]
[
  Transaction {
    tx_body: {
      operation: [Object],
      nonce: 0,
      gas_price: 1,
      timestamp: 1638364332668
    },
    signature: '0xb2a64bb8a15da85fc83ea523d395378ebb00b814b973b1b04b65e5fdbfbecc23cea8b9016d9a3107e230b9d15e0bf1e96e9b0284e9ca2aa3038ae77f0440d1ba6a801b9384a2ba82bacc8b373143e2cee8dbd49ba61ea1e7b9086ef0fa09efbc1c',
    hash: '0xb2a64bb8a15da85fc83ea523d395378ebb00b814b973b1b04b65e5fdbfbecc23',
    address: '0x00ADEc28B6a845a085e03591bE7550dd68673C1C',
    extra: { created_at: 1638364332672, executed_at: null }
  },
  Transaction {
    tx_body: {
      operation: [Object],
      nonce: 1,
      gas_price: 1,
      timestamp: 1638364332674
    },
    signature: '0x001c9336f349016432f8cff4fd5438b2dc52dd4feda9cfc2c39ee90ada670927e7e9bf0798b2308ae3094bd32d1ca8ec8193819f8c73fc52bb4a8b3cf831389e1f2ad679428172d1a7ee0e453f04706242c991e145cff2281165115fc8f9dad71b',
    hash: '0x001c9336f349016432f8cff4fd5438b2dc52dd4feda9cfc2c39ee90ada670927',
    address: '0x00ADEc28B6a845a085e03591bE7550dd68673C1C',
    extra: { created_at: 1638364332678, executed_at: null }
  },
  ...
  Transaction {
    tx_body: {
      operation: [Object],
      nonce: 10,
      gas_price: 1,
      timestamp: 1638364332706
    },
    signature: '0xd5cb4b150d1a8fe09780b11c757e941a11692edf6b80ece8ae86d18641b12607b7305baa59fabf3fa1d0f538d17aecaf0ed10b9db93b46376c96fb5db6e7a9526955b6523d2e6b3e9afcab226d9517ec8305827c8755a8e0326ffdf07ee067e21b',
    hash: '0xd5cb4b150d1a8fe09780b11c757e941a11692edf6b80ece8ae86d18641b12607',
    address: '0x00ADEc28B6a845a085e03591bE7550dd68673C1C',
    extra: { created_at: 1638364332710, executed_at: null }
  }
]
```

File: unittest/tx-pool.test.js (hunk 2 above)
Comment (shyun-comcom) [reply unittest/tx-pool.test.js]: In `Block.create()`, `undefined` timestamp is automatically set as `Date.now`.
(ref: https://github.com/ainblockchain/ain-blockchain/blob/master/blockchain/block.js#L87)

File: unittest/tx-pool.test.js (hunk 1 above)
Comment (platfowner) [reply unittest/tx-pool.test.js:632]: Thanks for the explanation!

**Pull request merged**